components/cdc/src/delegate.rs

```diff
@@ -423,22 +424,34 @@ impl Delegate {
         Some(resolved_ts)
     }
 
-    pub fn on_batch(&mut self, batch: CmdBatch) -> Result<()> {
+    pub fn on_batch<W: WriteBatch>(&mut self, batch: CmdBatch, wb: &mut W) -> Result<()> {
         // Stale CmdBatch, drop it sliently.
         if batch.observe_id != self.id {
             return Ok(());
         }
         for cmd in batch.into_iter(self.region_id) {
             let Cmd {
                 index,
-                mut request,
+                request,
                 mut response,
             } = cmd;
             if !response.get_header().has_error() {
-                if !request.has_admin_request() {
-                    self.sink_data(index, request.requests.into());
-                } else {
-                    self.sink_admin(request.take_admin_request(), response.take_admin_response())?;
+                match request {
+                    CmdRequest::PBCmdRequest(mut pb_request) => {
```

Comment:
Prefer `KvCmdRequest`. 🤔

Reply:
But they are both kv_requests....
The write-batch format only include key-value pairs while RaftCmdRequest may have an `AdminRequest`.